fluent-package/yum/fluent-package.spec.in

```diff
@@ -188,16 +198,33 @@ else
     fi
 fi
 if [ $1 -eq 2 ]; then
-  . %{_sysconfdir}/sysconfig/@SERVICE_NAME@
-  echo "pre FLUENT_PACKAGE_SERVICE_RESTART: $FLUENT_PACKAGE_SERVICE_RESTART"
-  if [ "$FLUENT_PACKAGE_SERVICE_RESTART" = auto ]; then
-    # collect installed gems during upgrading
+  # Collect plugin-list.
+  # Note: This should be done in %preun(1) of FROM-side, but we have no choice but to do this here
+  # because it needs to be done before installing files (it replaces embedded Ruby env).
```

Comment:
because it needs to be done before TO-side installing files but actually %preun(1) of FROM-side will be executed after it. (it replaces embedded Ruby env).